velero-plugin-for-aws/object_store.go

```diff
@@ -284,6 +328,10 @@ func (o *ObjectStore) PutObject(bucket, key string, body io.Reader) error {
 	case o.kmsKeyID != "":
 		req.ServerSideEncryption = aws.String("aws:kms")
 		req.SSEKMSKeyId = &o.kmsKeyID
+	// if sseCustomerKey is not empty, assume SSE-C encryption with AES256 algorithm
+	case o.sseCustomerKey != "":
+		req.SSECustomerAlgorithm = aws.String("AES256")
```

Comment:
Do we need to set `req.ServerSideEncryption = aws.String("AES256")` as well?

Reply:
No, SSE-C only requires the client to set the `SSECustomerAlgorithm` and `SSECustomerKey` fields. See https://pkg.go.dev/github.com/aws/aws-sdk-go-v2/service/s3#Client.GetObject (ctrl+f for `SSE-C`)